Cython/Compiler/Parsing.py

```diff
@@ -3009,17 +3009,20 @@ def p_with_gil(s):
     else:
         return 0
 
-def p_exception_value_clause(s):
-    # the default behaviour is to always
-    # check for exceptions
-    exc_check = 1
+def p_exception_value_clause(s, ctx):
     exc_val = None
+    if ctx.visibility  == 'extern':
+        exc_check = 0
+    else:
+        exc_check = 1
```

Comment:
Has there been some further discussion on consistent vs. less-breaking? By my count there wasn't a strong direction either way (both in this PR and the issue), but perhaps slightly in favour of consistency (i.e. default is to always check, also for extern functions)?

Reply:
Sorry -- I was really just experimenting here and trying to get tests to pass before making further progress either ways :)

Reply:
This was decided now in https://github.com/cython/cython/issues/4280#issuecomment-1100588580